### DISCUSSION TITLE
How to Obtain inboundPlanId and shipmentId for Shipments Created via Third-Party Applications in SP API v2024-03-20

### DISCUSSION THREAD

**Discussion by siriustigtr** [amzn/selling-partner-api-models#4275] [Q&A]
Currently, I'm retrieving information only through version v0. I make a request, receive the shipmentId, and based on that shipmentId, I get information about which SKUs I've sent and how many Amazon has received.

Now, when I try using the new version, almost all requests require an inboundPlanId and shipmentId, but I have no idea how to obtain this information. Logically, they are provided when I create shipments through the API. But what do I do in cases where shipments are created through third-party apps like 2D......., and I end up with nothing, so I can't make requests using the new version?

When I download all shipments—ACTIVE, VOIDED, or SHIPPED—they have no names, so I don't even understand which ones they are because the request returns an empty field for the name. Can someone help me with advice on how web applications like Seller......   handle this? I would be very grateful for any help.

**Comment by mafge** [comment]
Hey @siriustigtr,
thanks for reaching out!
In case you do not have any information about your shipments, you need to call [listInboundPlans](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#listinboundplans), then use the `inboundPlanId` to [getInboundPlan](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#getinboundplan) which gives you the `shipmentId(s)` for that plan which you can use to get further information about the shipment via [getShipment](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#getshipment) operation. In order to distinguish shipments you can use [listShipmentItems](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#listshipmentitems) to check what is in the shipment, additionally you can use [updateShipmentName](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#updateshipmentname) to set a custom name.

This is not very straightforward to be honest, but usually sellers use a solution that covers the whole flow without the need to jump in at some point in between. Can you share some more details on your use case, maybe there is some other way that I don't see at the moment with the given information. Hope that helps!
Best,
Marc
Selling Partner Developer Services

**Reply by siriustigtr** [reply]
Hi,
 When I call listInboundPlans, I receive a large number of inbound IDs, each with one of three statuses: ACTIVE, VOIDED, or SHIPPED. However, shipments with statuses like CLOSED or RECEIVING in Seller Central don’t appear under ACTIVE or SHIPPED. This means I only get inbound IDs for shipments that are not yet fully finalized. Additionally, none of these inbound IDs return a name, so they have no names assigned. I hope this helps clarify my issue.

**Reply by mafge** [reply]
Hi @siriustigtr,
thanks for providing this information. In your case I would recommend to stick to v0 operations for now as there is no proper replacement in the new API available.
